java/src/main/native/src/RmmJni.cpp

```diff
@@ -502,46 +365,47 @@ JNIEXPORT void JNICALL Java_ai_rapids_cudf_Rmm_initializeInternal(JNIEnv *env, j
     bool use_pool_alloc = allocation_mode & 1;
     bool use_managed_mem = allocation_mode & 2;
     if (use_pool_alloc) {
-        std::vector<int> devices; // Just do default devices for now...
-        if (use_managed_mem) {
-            auto tmp =  new rmm::mr::cnmem_managed_memory_resource(pool_size, devices);
-            Initialized_resource.reset(tmp);
-            auto wrapped = make_tracking_adaptor(tmp, RMM_ALLOC_SIZE_ALIGNMENT);
-            Tracking_memory_resource.reset(wrapped);
-        } else {
-            auto tmp =  new rmm::mr::cnmem_memory_resource(pool_size, devices);
-            Initialized_resource.reset(tmp);
-            auto wrapped = make_tracking_adaptor(tmp, RMM_ALLOC_SIZE_ALIGNMENT);
-            Tracking_memory_resource.reset(wrapped);
-        }
-    } else if (rmm::Manager::useManagedMemory()) {
-        auto tmp =  new rmm::mr::managed_memory_resource();
+      std::vector<int> devices; // Just do default devices for now...
+      if (use_managed_mem) {
+        auto tmp = new rmm::mr::cnmem_managed_memory_resource(pool_size, devices);
         Initialized_resource.reset(tmp);
         auto wrapped = make_tracking_adaptor(tmp, RMM_ALLOC_SIZE_ALIGNMENT);
         Tracking_memory_resource.reset(wrapped);
-    } else {
-        auto tmp =  new rmm::mr::cuda_memory_resource();
+      } else {
+        auto tmp = new rmm::mr::cnmem_memory_resource(pool_size, devices);
         Initialized_resource.reset(tmp);
         auto wrapped = make_tracking_adaptor(tmp, RMM_ALLOC_SIZE_ALIGNMENT);
         Tracking_memory_resource.reset(wrapped);
+      }
+    } else if (rmm::Manager::useManagedMemory()) {
```

Comment:
Sorry, I just noticed this is a bug that I introduced.  Could you switch this to 
```
} else if (use_managed_mem) {
```

Reply:
Done.